trillian/ctfe/handlers.go

```diff
@@ -456,13 +458,35 @@ func addChainInternal(ctx context.Context, li *logInfo, w http.ResponseWriter, r
 	} else if len(rest) > 0 {
 		return http.StatusInternalServerError, fmt.Errorf("extra data (%d bytes) on reconstructing MerkleTreeLeaf", len(rest))
 	}
+	sctVersion := ct.V1
+	if li.misbehave("use submitted leaf not already-logged leaf (if present) in SCT") {
```

Comment:
Does this imply a n*(1/2, 1/4, 1/8) probability across these types of misbehavior? (passim)
 
maybe  something like
```golang
type misbehaviour struct {
  desc string
  playup func()
}

func (li *...) chooseToBeBad(behaviours []behaviour) {
  if !badlog {return}

  if rand(100) != 0 { return } 
  b := misbehaviours[rand(len(misbehaviours))]
  log.INFO("DELIBERATE: %s", b.desc)
  b.playUp()
}


...
   ...
   li.chooseToBeBad([]behaviours{
     { "use submitted leaf..", func() { ... } },
     { "change SCT timestamp...", func() { ... }}.
     ...
      {...})
```

or similar?

Reply:
It does imply a reduced chance for later misbehaviour, but it's a (0.99)<sup>N-1</sup>/100 chance for the N-th arm:
 - first arm has a 0.01 (=1/100) chance of happening
 - second arm has a 0.0099 (=99/100 * 1/100) chance
 - third arm has a 0.009801 (=99/100 * 99/100 * 1/100) chance
 - fourth arm has a 0.009703 (=99/100 * 99/100 * 99/100 * 1/100) chance
 - fifth arm has a 0.009606 (=99/100 * 99/100 * 99/100 * 99/100 * 1/100) chance.

So with a fixed misbehaviour rate of 1% I don't think the scaling is worth worring about.  If the rate were configurable, then I agree that something more sophisticated would be worthwhile, but I'm generally trying to keep this set of diffs as simple as possible (given the intent of maintaining a separate branch).